docs/proposals/20230123-azure-service-operator.md

```diff
@@ -0,0 +1,172 @@
+  - Multi- or single-tenant?
+  - Install in capz-controller-manager `func main()`
+- GET requests are not as straightforward in the ASO model because BYO resources
+  may not initially exist in ASO
+- Current service pattern in CAPZ works for Azure services that are not
+  represented in ASO (resource health)
+- Each release of ASO can expose several Azure API versions of each resource,
+  including GA and preview API versions.
+
+### Security Model
+
+- ASO requires credentials to interact with Azure, so CAPZ will have to ensure those are securely passed on to ASO if it is not already installed
+- ASO resources are subject to Kubernetes RBAC => capz-controller-manager will require many more permissions that may extend beyond what's minimally required.
+  - If a user already has their own ASO resources installed, CAPZ may require
+    permission to operate against them to handle the ASO resources it needs to
+    manage.
+
+### Risks and Mitigations
+
+- Introduction of functional regressions
+- Introduction of performance regressions, where operations against workload
+  clusters may take longer to complete
+- Introduction of scalability regressions, where management clusters may exhaust
+  Azure or Kubernetes API limits with fewer or smaller workload clusters being managed.
+  - Management cluster will have to manage many more Kubernetes resources per
+    workload cluster
+- Because ASO has not yet been proven as a mission-critical interface to Azure
+  to the same extent as Azure's REST API and Go SDK, there is an apparent increased risk
+  that CAPZ's lifespan may extend beyond that of its Azure interface when ASO is
+  used instead of the API or SDK directly
+- Conflicting user installations of ASO or ASO resources
+- Future breaking changes in ASO
+- Lower-fidelity telemetry compared to what CAPZ tracks currently
+
+## Alternatives
+
+### Adopt new SDK directly
+
+To shed its dependency on the deprecated Azure Go SDK, CAPZ could adopt the new Azure Go SDK and use it directly instead of transitively through ASO. The newer SDK's concepts like how Azure resources are defined and operated against map much more directly to those of the older SDK, so a transition to using the newer SDK directly is likely much more mechanical than a similar move to ASO. A simpler transition would not be as prone to many of the risks listed above, particularly the risk of introducing regressions. It would also likely take less time to complete, mitigating any risks and reducing maintenance costs associated with using both the old and new patterns at the same time in difference service implementations.
+
+### Crossplane
+
+Another alternative is [Crossplane](https://www.crossplane.io/), a CNCF project which is a set of Kubernetes controllers and Custom Resource Definitions like ASO that define and operate on Azure resources. A number of factors indicate that ASO may be a better fit for CAPZ:
+- Maintained by [Upbound](https://www.upbound.io/), not Microsoft
```

Comment:
I don't think that's really a valid "pro"

Reply:
Removed